### PR TITLE
Fix GROUPING function SQL compatibility

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Make the result of GROUPING function the same as in SQL and other DBMS.

### CODE REVIEW

File: src/Functions/grouping.h [hunk 1]
```diff
@@ -19,10 +19,15 @@ class FunctionGroupingBase : public IFunction
     static constexpr UInt64 ONE = 1;
 
     const ColumnNumbers arguments_indexes;
+    const bool force_compatibility;
```
Comment (nickitat) [inline src/Functions/grouping.h:22]: in context of this pr it is clear what `compatibility` you're referring to, but for the future reader it probably will be not easy to get. add pls a brief comment

File: src/Functions/grouping.h [hunk 2]
```diff
@@ -48,13 +53,15 @@ class FunctionGroupingBase : public IFunction
         auto result = ColumnUInt64::create();
         auto & result_data = result->getData();
         result_data.reserve(input_rows_count);
+
+        const auto * result_table = likely(force_compatibility) ? COMPATIBLE_MODE : INCOMPATIBLE_MODE;
```
Comment (nickitat) [inline src/Functions/grouping.h:57]: not an issue, but imo we should limit use of this intrinsic to hot loops when we also for some reason don't trust hw predictor

File: src/Functions/grouping.h (hunk 1 above)
Comment (novikd) [inline src/Functions/grouping.h:22]:
```suggestion
    // Initial implementation of GROUPING function returned 1 if the argument is used as an aggregation key.
    // This differs from the behavior described in the standard and other DBMS.
    const bool force_compatibility;
```

**Review by nickitat — APPROVED** [review]

**Comment by den-crane** [comment]
spent sometime on debuging it, somehow it is not in changelog

```
clickhouse/docs/changelogs

grep -R 40762 *
<nothing>
```

**Pull request merged**